docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md

```diff
@@ -0,0 +1,158 @@
+
+```zsh
+git clone https://github.com/ava-labs/avalanche-smart-contract-quickstart.git
+cd avalanche-smart-contract-quickstart
+yarn
+```
+
+In order to deploy contracts, you need to have some AVAX. You can use a pre-funded account on a local network or get testnet AVAX from the [Avalanche Faucet](https://faucet.avax.network), which is an easy way to get to play around with Avalanche. After getting comfortable with your code, you can run it on Mainnet after making the necessary changes to your workflow.
+
+## Implement a Game Item NFT
+
+We will use our example ERC721 smart contract, [`NFT.sol`](https://github.com/ava-labs/avalanche-smart-contract-quickstart/blob/3fbba0ac28f6420e9be5d2635d5f23693f80127a/contracts/NFT.sol) found in `./contracts` of our project.
+
+```ts
+//SPDX-License-Identifier: MIT
+// contracts/ERC721.sol
+
+pragma solidity >=0.6.2;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+contract NFT is ERC721 {
+  using Counters for Counters.Counter;
+  Counters.Counter private _tokenIds;
+
+  constructor() ERC721("GameItem", "ITM") {}
+
+  // commented out unused variable
+  // function awardItem(address player, string memory tokenURI)
+  function awardItem(address player)
+    public
+    returns (uint256)
+  {
+    _tokenIds.increment();
+
+    uint256 newItemId = _tokenIds.current();
+    _mint(player, newItemId);
+    // _setTokenURI(newItemId, tokenURI);
+
+    return newItemId;
+  }
+}
+```
+
+Let's examine this implementation of an NFT as a Game Item. We start by importing to contracts from our node modules. We import Openzeppelin's open source implementation of the ERC721 standard which our NFT contract will inherit from. Our constructor takes the `_name` and `_symbol` arguments for our NFT and passes them on to the constructor of the parent ERC721 implementation. Lastly we implement the `awardItem` function which allows anyone to mint an NFT to a player's wallet address. This function increments the `currentTokenId` and makes use of the `_mint` function of our parent contract.
+
+# Compile & deploy with forge
+
+To compile the NFT contract run:
+
+ ```zsh
+ forge build
+ ``` 
+ 
+ By default the compiler output will be in the `out` directory. To deploy our compiled contract with Forge we have to set environment variables for the RPC endpoint and the private key we want to use to deploy.
+
+Set your environment variables by running:
+
+```zsh
+export RPC_URL=<YOUR-RPC-ENDPOINT>
+export PRIVATE_KEY=<YOUR-PRIVATE-KEY>
+```
+
+Since we are deploying to Fuji testnet, our `RPC_URL` export should be:
+
+```zsh
+export RPC_URL=https://api.avax-test.network/ext/bc/C/rpc
+```
+
+Once set, you can deploy your NFT with Forge by running the command below while adding the relevant constructor arguments of the NFT contract:
+
+```zsh
+forge create NFT --rpc-url=$RPC_URL --private-key=$PRIVATE_KEY --constructor-args <NFT-NAME> <NFT-SYMBOL> 
```

Comment:
Where are the value of < NFT-NAME > and < NFT-SYMBOL > from and how are they related to the code above?

Reply:
Once set, you can [deploy your NFT with Forge](https://book.getfoundry.sh/reference/forge/forge-create.html) by running the command below while adding the values for `_name` and `_symbol`, the relevant [constructor arguments](https://github.com/ava-labs/avalanche-smart-contract-quickstart/blob/3ad93abf50fba65e3aab68f23382bcace73968be/contracts/NFT.sol#L13) of the NFT contract:

```zsh
forge create NFT --rpc-url=$RPC_URL --private-key=$PRIVATE_KEY --constructor-args GameItem ITM 
```

Hows this?